var/spack/repos/builtin/packages/libbsd/package.py

```diff
@@ -32,19 +32,11 @@ class Libbsd(AutotoolsPackage):
 
     patch('cdefs.h.patch', when='@0.8.6 %gcc@:4')
     patch('local-elf.h.patch', when='@:0.10 %intel')
-    patch('nvhpc.patch', when='%nvhpc')
```

Comment:
This should be kept, but limited to `when='@:0.10 %nvhpc'`

Reply:
Actually, tracing back through this, `nvhpc.patch` was required because `patch()` modifies configure, and thus triggers an autoreconf.  The autoreconf causes `src/Makefile.in` to get regenerated, which replaced `-include ../config.h` with `-I ../config.h`.  `nvhpc.patch` worked around that issue by explicitly including `config.h` in the affected source files.

Since the flags that `patch()` was removing have been supported in the HPC SDK since version 21.7, it's probably safe to remove both `patch()` and `nvhpc.patch`.